Lib/test/test_ast/test_ast.py

```diff
@@ -1614,6 +1614,12 @@ def test_literal_eval_syntax_errors(self):
                 (\
             \ ''')
 
+    def test_literal_eval_large_input_crash(self):
+        # gh-125010: Fix use-after-free in ast repr()
+        source = "{0x0" + "e" * 250_000 + "%" + "e" * 250_000 + "1j}"
+        with self.assertRaisesRegex(ValueError, "Exceeds the limit"):
+            ast.literal_eval(source)
```

Comment:
Note for the other reviewers:
Without setting `sys.set_int_max_str_digits(0)` this code throws a `ValueError: Exceeds the limit (4300 digits)`. With `sys.set_int_max_str_digits(0)` this code throws a `ValueError: malformed node or string on line 1:`

Reply:
Maybe it'd be good to include such comment at the source level?

Reply:
> Maybe it'd be good to include such comment at the source level?

Agreed!

Reply:
I added a note to the test :)